### PR TITLE
chore: update tests to use blueprint test framework

### PR DESCRIPTION
**Pull request by bharathkkb**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_documentation
 The documentation needs to be regenerated. Please run `make generate_docs`. 
 ```bash 
Checking for documentation generation
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/examples/minimal/README.md /tmp/tmp.RnOaQRMh9m/generate_docs/workspace/examples/minimal/README.md
10,12d9
< | location\_id | Zone to create test instance. | `string` | `"us-east1-b"` | no |
< | memory\_size\_gb | Memory size of test instance. | `number` | `1` | no |
< | name | Name of redis instance. | `string` | `"test-minimal"` | no |
14,15d10
< | region | Region to create test instance. | `string` | `"us-east1"` | no |
< | tier | The service tier of the instance. https://cloud.google.com/memorystore/docs/redis/reference/rest/v1/projects.locations.instances#Tier | `string` | `"BASIC"` | no |
21,23d15
< | location\_id | n/a |
< | memory\_size\_gb | n/a |
< | name | n/a |
29d20
< | region | n/a |
diff -r '--exclude=.terraform' '--exclude=.kitchen' '--exclude=.git' '--exclude=autogen' '--exclude=*.tfvars' /workspace/examples/redis/README.md /tmp/tmp.RnOaQRMh9m/generate_docs/workspace/examples/redis/README.md
10,15d9
< | alternative\_location\_id | Alternate Zone to create test instance. | `string` | `"us-east1-d"` | no |
< | auth\_enabled | Indicates whether OSS Redis AUTH is enabled for the instance. If set to true AUTH is enabled on the instance. | `bool` | `true` | no |
< | auth\_string | AUTH String set on the instance. This field will only be populated if auth\_enabled is true | `string` | `null` | no |
< | location\_id | Zone to create test instance. | `string` | `"us-east1-b"` | no |
< | memory\_size\_gb | Memory size of test instance. | `number` | `1` | no |
< | name | Name of redis instance. | `string` | `"test-redis"` | no |
17,18d10
< | region | Region to create test instance. | `string` | `"us-east1"` | no |
< | transit\_encryption\_mode | The TLS mode of the Redis instance, If not provided, TLS is disabled for the instance. | `string` | `"SERVER_AUTHENTICATION"` | no |
24,29d15
< | alternative\_location\_id | n/a |
< | auth\_enabled | n/a |
< | auth\_string | n/a |
< | location\_id | n/a |
< | memory\_size\_gb | n/a |
< | name | n/a |
36,37d21
< | region | n/a |
< | transit\_encryption\_mode | n/a |
Error: Documentation generation has not been run, please run the
'make docker_generate_docs' command and commit the above changes.
 ``` 
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
Checking for missing newline at end of file
Error: No newline at end of file ./examples/memcache/network.tf
Error: No newline at end of file ./examples/redis/network.tf
Error: No newline at end of file ./examples/minimal/network.tf
 ``` 
- ⚠️check_headers
 All files need a license header. Please make sure all your files include the appropriate header. A helper tool available [here](https://github.com/google/addlicense). 
 ```bash 
Checking file headers
./test/integration/memcache/memcache_test.go
./test/integration/redis/redis_test.go
./test/integration/discover_test.go
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
examples/memcache/main.tf
--- old/examples/memcache/main.tf
+++ new/examples/memcache/main.tf
@@ -25,13 +25,13 @@
 }
 
 module "memcache" {
-  source         = "../../modules/memcache"
-  name           = "example-memcache"
-  project        = var.project_id
-  memory_size_mb = "1024"
-  enable_apis    = true
-  cpu_count      = "1"
-  region         = "us-east1"
+  source             = "../../modules/memcache"
+  name               = "example-memcache"
+  project            = var.project_id
+  memory_size_mb     = "1024"
+  enable_apis        = true
+  cpu_count          = "1"
+  region             = "us-east1"
   authorized_network = module.test-vpc-module.network_id
   depends_on = [
     module.private-service-access.peering_completed
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
examples/redis/main.tf
--- old/examples/redis/main.tf
+++ new/examples/redis/main.tf
@@ -26,6 +26,6 @@
   enable_apis             = true
   auth_enabled            = true
   transit_encryption_mode = "SERVER_AUTHENTICATION"
-authorized_network = module.test-vpc-module.network_id
-  memory_size_gb = 1
+  authorized_network      = module.test-vpc-module.network_id
+  memory_size_gb          = 1
 }
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
examples/redis/outputs.tf
--- old/examples/redis/outputs.tf
+++ new/examples/redis/outputs.tf
@@ -35,6 +35,6 @@
 }
 
 output "output_auth_string" {
-  value = module.memstore.auth_string
+  value     = module.memstore.auth_string
   sensitive = true
 }
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
examples/minimal/main.tf
--- old/examples/minimal/main.tf
+++ new/examples/minimal/main.tf
@@ -17,12 +17,12 @@
 module "memstore" {
   source = "../.."
 
-  name = "test-minimal"
-  project     = var.project_id
-  region      = "us-east1"
-  location_id = "us-east1-b"
-  enable_apis = true
-  tier        = "BASIC"
+  name               = "test-minimal"
+  project            = var.project_id
+  region             = "us-east1"
+  location_id        = "us-east1-b"
+  enable_apis        = true
+  tier               = "BASIC"
   authorized_network = module.test-vpc-module.network_id
 
   memory_size_gb = 1
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by JulienBreux** [comment]
Thanks @bharathkkb very good news here!

File: test/integration/redis/redis_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	bpt := tft.NewTFBlueprintTest(t)
+
+	bpt.DefineVerify(func(assert *assert.Assertions) {
+		bpt.DefaultVerify(assert)
+
+		projectId := bpt.GetStringOutput("project_id")
+
+		op := gcloud.Runf(t, "redis instances describe test-redis --project=%s --region=us-east1", projectId)
+		assert.True(op.Get("authEnabled").Bool())
+		assert.Contains(op.Get("memorySizeGb").String(), "1")
```
Comment (morgante) [inline test/integration/redis/redis_test.go:35]: These should probably be assert.Equal. We wouldn't want a value like "10" to pass.

**Review by morgante — APPROVED** [review]

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: test/integration/redis/redis_test.go (hunk 1 above)
Comment (bharathkkb) [reply test/integration/redis/redis_test.go]: good catch, updated for both

**Pull request merged**